### PR TITLE
Antenna update only

### PR DESCRIPTION
**Pull request by Andrewsopoku**
**Issue**

- Link: https://github.com/NebraLtd/maker-starter-app/issues/117
- Summary: PR for updating only antenna

**How**

**Screenshots**

Simulator Screen Shot - iPhone 13 - 2022-05-02 at 11 30 59

Simulator Screen Shot - iPhone 13 - 2022-05-02 at 11 31 07

**References**

**Checklist**

- [ ] Tests added
- [ ] Cleaned up commit history (rebase!)
- [ ] Documentation added
- [ ] Thought about variable and method names

### CODE REVIEW

File: src/features/hotspots/setup/HotspotSetupConfirmLocationScreen.tsx [hunk 1]
```diff
@@ -80,7 +86,17 @@ const HotspotSetupConfirmLocationScreen = () => {
       onboardingRecord,
       dataOnly: false,
     }
-
+    if (params.updateAntennaOnly) {
+      const { fee, stakingFee } = calculateAssertLocFee(
```
Comment (marvinmarnold) [inline src/features/hotspots/setup/HotspotSetupConfirmLocationScreen.tsx:90]: Can you add some comments to explain why undefined is being supplied

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx [hunk 2]
```diff
@@ -107,8 +115,38 @@ const HotspotTxnsProgressScreen = () => {
...
+      const assertLocationTxn = await Location.createLocationTxn({
+        gateway: hotspotAddress,
+        lat: hotspot.lat,
+        lng: hotspot.lng,
+        decimalGain: params.gain,
+        elevation: params.elevation,
+        dataOnly: false,
+        owner: ownerAddress,
+        currentLocation,
+        makerAddress: onboardingRecord?.maker?.address || '',
```
Comment (marvinmarnold) [inline src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:148]: Should this default to Nebra's maker address?

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx [hunk 3]
```diff
@@ -107,8 +115,38 @@ const HotspotTxnsProgressScreen = () => {
...
+              text: t('hotspot_setup.antenna_only_fee.no_location_asserted.ok'),
+              onPress: () => navigation.navigate('MainTabs'),
+            },
+          ],
+        )
+
+        return
+      }
+      const currentLocation = getH3Location(hotspot.lat, hotspot.lng)
+      const assertLocationTxn = await Location.createLocationTxn({
```
Comment (marvinmarnold) [inline src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:139]: I think you've already said this, but can you confirm that this method will not charge a fee as long as the lat and long don't change?

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx [hunk 4]
```diff
@@ -77,15 +79,20 @@ const HotspotTxnsProgressScreen = () => {
 
     // check if add gateway needed
     const isOnChain = await hotspotOnChain(hotspotAddress)
-    const onboardingRecord = await getOnboardingRecord(hotspotAddress)
-    if (!onboardingRecord) return
+
+    let onboardingRecord
+    try {
+      onboardingRecord = await getOnboardingRecord(hotspotAddress)
+    } catch (error) {}
```
Comment (marvinmarnold) [inline src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:86]: Why are we swallowing the error here? We don't want to notify sentry or take any other action?

**Review by marvinmarnold — COMMENTED** [review]
Do I understand correctly that following these changes, users can either:

- update location and antenna
- update antenna only

But there is no option to update location only?

**Comment by Andrewsopoku** [comment]
> Do I understand correctly that following these changes, users can either:
> 
> * update location and antenna
> * update antenna only
> 
> But there is no option to update location only?

Yes. Updating  "location and antenna" and  "location only" will come with the same cost.

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx (hunk 3 above)
Comment (Andrewsopoku) [reply src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:139]: Yes. The H3 Location.

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx (hunk 2 above)
Comment (Andrewsopoku) [reply src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:148]: This takes care of Non Nebra hotspots. The makerAddress param is used in determining the payer. If an empty string is supplied, the owner becomes the payer.

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx (hunk 4 above)
Comment (marvinmarnold) [reply src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:86]: Gotcha, please add comments explaining this.

**Review by marvinmarnold — APPROVED** [review]

**Pull request merged**